scripts/tests/sanitycheck/test_reporting_testsuite.py

```diff
@@ -0,0 +1,168 @@
+    Test 2: Assert on the columns and values of the generated csv_report"""
+    class_testsuite.instances = instances_fixture
+    filename = tmpdir.mkdir("test_csv").join("sanitycheck_csv_report.csv")
+    class_testsuite.csv_report(filename)
+    assert os.path.exists(filename)
+    assert os.stat(filename).st_size != 0
+
+    mydict = {'test': [], 'arch' : [], 'platform' : [], 'status': [],
+              'extra_args': [], 'handler': [], 'handler_time': [],
+              'ram_size': [], 'rom_size': []}
+
+    with open(filename, "r") as file:
+        csv_reader = csv.reader(file)
+        assert set(mydict.keys()) == set(list(csv_reader)[0])
+
+    for instance in class_testsuite.instances.values():
+        mydict["test"].append(instance.testcase.name)
+        mydict["arch"].append(instance.platform.arch)
+        mydict["platform"].append(instance.platform.name)
+        instance_status = instance.status if instance.status is not None else ""
+        mydict["status"].append(instance_status)
+        args = " ".join(instance.testcase.extra_args)
+        mydict["extra_args"].append(args)
+        mydict["handler"].append(instance.platform.simulation)
+        mydict["handler_time"].append(instance.metrics.get("handler_time", ""))
+        mydict["ram_size"].append(instance.metrics.get("ram_size", '0'))
+        mydict["rom_size"].append(instance.metrics.get("rom_size", '0'))
+
+    dict_file = open(filename, "r")
+    dict_reader = csv.DictReader(dict_file)
+    columns = defaultdict(list)
+    for row in dict_reader:
+        for (key, value) in row.items():
+            columns[key].append(value)
+    for i in range(len(mydict)):
+        assert set(list(columns.values())[i]) == set(list(mydict.values())[i])
+    dict_file.close()
+
+def test_xunit_report(class_testsuite, test_data,
+                      instances_fixture, platforms_list, all_testcases_dict):
+    """ Testing xunit_report function of Testsuite class in sanitycheck
+    Test 1: Assert sanitycheck.xml file exists after execution of xunit_report function
+    Test 2: Assert on fails, passes, skips, errors values
+    Test 3: Assert on the tree structure of sanitycheck.xml file"""
+    class_testsuite.platforms = platforms_list
+    class_testsuite.testcases = all_testcases_dict
+    kwargs = {"exclude_tag" : ['test_a'], "exclude_platform" : ['demo_board_1'],
+              "platform" : ['demo_board_2']}
+    class_testsuite.apply_filters(**kwargs)
+    class_testsuite.instances = instances_fixture
+    inst1 = class_testsuite.instances.get("demo_board_2/scripts/tests/sanitycheck/test_data/testcases/tests/test_a/test_a.check_1")
+    inst2 = class_testsuite.instances.get("demo_board_2/scripts/tests/sanitycheck/test_data/testcases/tests/test_a/test_a.check_2")
+    inst1.status = "failed"
+    inst2.status = "skipped"
+
+    filename = test_data + "sanitycheck.xml"
+    class_testsuite.xunit_report(filename)
+    assert os.path.exists(filename)
```

Comment:
testing if a file exists is not enough, it might have garbage in it or might be empty, we just had a bug like this in mainline tree.

Reply:
Yes agree just testing if filename is not enough, but the second part of this testcase checks for the content of the sanitycheck.xml report as well. Let me know if you have any suggestions for adding more checks. Thanks.